workers/common/src/main/kotlin/common/OrtRunService.kt

```diff
@@ -100,6 +101,20 @@ class OrtRunService(
 ) {
     companion object {
         private const val RUN_ID_LABEL = "runId"
+
+        /**
+         * Validate this [WorkerJob] before it can be processed by the corresponding worker. Check whether the job
+         * exists and has a state that allows it to be processed.
+         */
+        inline fun <reified T : WorkerJob> T?.validateForProcessing(jobId: Long): T {
+            requireNotNull(this) { "The ${T::class.simpleName} '$jobId' does not exist in the database." }
+
+            if (status.final) {
+                throw JobIgnoredException("The job is in a final state.")
```

Comment:
Should you also put the worker's class name and the job id in the exception message ?

Reply:
Yes, I intended to do this, but got distracted. Fixed now.